### PR TITLE
task: update typings for @mapbox/mapbox-sdk

### PR DESCRIPTION
**Pull request by theworkflow**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://docs.mapbox.com/api/maps/#create-an-upload>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/mapbox__mapbox-sdk/index.d.ts [hunk 1]
```diff
@@ -1699,7 +1700,7 @@ declare module '@mapbox/mapbox-sdk/services/uploads' {
          * Create an upload.
          * @param config
          */
-        createUpload(config: { mapId: string; url: string; tilesetName?: string }): MapiRequest;
+        createUpload(config: { mapId: string; url: string; tilesetName?: string, name?: string, private?: boolean }): MapiRequest;
```
Comment (jeffbdye) [inline types/mapbox__mapbox-sdk/index.d.ts:1703]: I'm not sure that I see a `private` parameter on the config object passed into `createUpload` here, is that intentional? Or is this coming in a new version of the mapbox-sdk?

https://github.com/mapbox/mapbox-sdk-js/blob/fc74a74f6c48d7373aec06548379f0c534f4d298/services/uploads.js#L121

**Comment by jeffbdye** [comment]
Additionally, the build failure might mean we need to revisit how these types are exported or perhaps see if it's alright to override due to how the mapbox services are used.

Updating export approach: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require

Or overriding the rule: 
> you can disable this check by adding the following options to your project's tslint.json file under "rules": 
> "npm-naming": [true,{"mode":"code","errors":[["NeedsExportEquals",false]]}]
>  See: https://github.com/Microsoft/dtslint/blob/master/docs/npm-naming.md

**Review by jeffbdye — CHANGES_REQUESTED** [review]
Sorry that the build failure came up with your changes, maybe we should see if we can get a DefinitelyTyped maintainer to suggest what path we take.

File: types/mapbox__mapbox-sdk/index.d.ts (hunk 1 above)
Comment (theworkflow) [reply types/mapbox__mapbox-sdk/index.d.ts:1703]: Looks like the SDK needs an update as well. [Here](https://docs.mapbox.com/api/maps/#create-an-upload) are the official docs from Mapbox where they support this. I'll also create a PR for the sdk to include the parameter

Comment (theworkflow) [reply types/mapbox__mapbox-sdk/index.d.ts:1703]: Once https://github.com/mapbox/mapbox-sdk-js/pull/373 gets merged I'll update the code here and the Mapbox sdk version

**Comment by jeffbdye** [comment]
@jessetrinity Sorry to tag you directly, but I believe you'd helped out when we originally contributed these types - any guidance on how we should proceed with the build failure on the types here? I feel like it's the latter approach mentioned due to the way the underlying library is a default export, but wanted to confirm with someone better versed with types like this.

We're still waiting on @theworkflow 's other PR to go through exposing another property he's adding here, so no rush on a response (see https://github.com/mapbox/mapbox-sdk-js/pull/373). Also let us know if there's someone else we should reach out to/tag instead if you're no longer a contact here!

> Additionally, the build failure might mean we need to revisit how these types are exported or perhaps see if it's alright to override due to how the mapbox services are used.
> 
> Updating export approach: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
> 
> Or overriding the rule:
> 
> > you can disable this check by adding the following options to your project's tslint.json file under "rules":
> > "npm-naming": [true,{"mode":"code","errors":[["NeedsExportEquals",false]]}]
> > See: https://github.com/Microsoft/dtslint/blob/master/docs/npm-naming.md

**Pull request closed without merging**